nifi-commons/nifi-site-to-site-client/src/main/java/org/apache/nifi/remote/client/SiteToSiteClient.java

```diff
@@ -560,7 +560,8 @@ public SiteToSiteClient build() {
         }
 
         /**
-         * @return the configured URL for the remote NiFi instance
+         * @return the configured URL for the remote NiFi instance.
+         * The URL string possibly contains multiple URL entries in comma-separated format.
          */
         public String getUrl() {
```

Comment:
@markap14 Thanks for the comment. I searched the usage of this method, SiteToSiteClientConfig is used to specify Site-to-Site configuration by modules that uses Site-to-Site, such as RemoteGroupPort, NiFiStormTopology, NiFiReceiver (for Spark). This getUrl() method is currently used from only nifi-site-to-site-client module to create instances of SiteToSiteClient, and those are updated to support multiple URLs by this PR.

I think this SiteToSiteClientConfig is set by other modules, but is only read by SiteToSiteClient. Even if other module uses getUrl(), the URL would be the one that is set by the module itself, so it should be fine. SiteToSiteClient also expose SiteToSiteClientConfig by getConfig() but it's only used by org.apache.nifi.remote.client.socket.SocketClient so far.

I think it's safe to change the meaning of this getUrl() method. But I'd add separate method if there's any other reason we should do so.